### ISSUE TITLE
Oned compaction breaks with new orthogonal self-loops

### ISSUE THREAD

**Issue by uruuru** [eclipse/elk#542]
Applying `LEFT` compaction to the
[selfloop_orthogonal.elkt](https://github.com/eclipse/elk-models/blob/master/tests/layered/compaction_oned/selfloop/selfloop_orthogonal.elkt) test file yields 
a) invalid hitboxes if the nodes have no particularly specified size
b) a broken layout (all nodes and edges stacked to the very left) if all nodes have size 20, 20

As far as I remember, the graph worked with the old self loops.

**Comment by uruuru** [comment]
I checked using elk live with different java versions. The issue occurs starting with 0.6.0. 

@le-cds, 0.5.0 already includes "new splines" right? Anything else that changed from 0.5.0 to 0.6.0 w.r.t. self loops?

**Comment by le-cds** [comment]
@uruuru [0.6.0](https://www.eclipse.org/elk/downloads/releasenotes/release-0.6.0.html) shipped my rewritten self loop code. Regarding 0.5.0, what exactly do you mean by new splines?

**Comment by uruuru** [comment]
The self loops in 0.5.0 and 0.6.0 look roughly the same. Maybe 0.6.0 is "your" new code :)? We had the results of a master's thesis in there before, right?

**Comment by le-cds** [comment]
Yep, you're right. 0.5.0 shipped the master's thesis code. My rewrite shipped with 0.6.0, but it did not alter the major concepts.
